### DISCUSSION TITLE
QR-Code Scanner and \r\n in QR-Code

### DISCUSSION THREAD

**Discussion by oliverlippold** [mebjas/html5-qrcode#273] [Ideas]
Hi,

the QR-Code Scanner is greate !!! The only think what I am missing or not found is that the scanner can't interprete \r\n (Linefeed) in ${decodedText}. So for example a QR-Code with:

adb\r\nbcd will not view as 

abc
def

it will viewed as

abc def

mayby you have a Idee ? ;-) :-)

BR
Oliver

**Comment by ROBERT-MCDOWELL** [comment]
maybe urlencode it!

**Comment by oliverlippold** [comment]
No, does not work, I do not get \r\n in ${decodedText} :-(, I only get a space, there is no other character.

Thank's for help again !

BR
Oliver

**Comment by ROBERT-MCDOWELL** [comment]
base64 encode so...

**Comment by oliverlippold** [comment]
Hi Robert,

can you send me a example ?, I do not understand at the moment your message :-(

Thanks a lot :-) !!!

Oliver

**Comment by ROBERT-MCDOWELL** [comment]
before to create your qrcode, encode the data with urlencode or base64encode.
then scan it with html5-qrcode and decode it.... simple.

**Comment by oliverlippold** [comment]
Hi Robert,

thanks for your tip but ist does not work with encoding the qrcode before generating and decoding after scanning. The problem is that the qrcode generator need a sting like this "PART1 LINE1\r\nLINE2\r\nLINE3\r\nLINE4", the result is a qrcode which can read from every smartphone correct (iPhone etc.). If I scan the qrcode with html5-qrcode i get the result in ${decodedText}, so it is decodeed and I get the Text as "PART1 LINE1 LINE2 LINE3 LINE4" in the browser, on a smartphone I get the output as:
PART1 LINE1
LINE2 
LINE3 
LINE4

So the Result is decodeed from html5-qrcode but there is no linefeed :-(

Here is my code from html5-qrcode:

</body>
<script src="/scripts/html5-qrcode-master/dist/html5-qrcode.min.js"></script>
<script>
    function docReady(fn) {
        // see if DOM is already available
        if (document.readyState === "complete"
            || document.readyState === "interactive") {
            // call on next available tick
            setTimeout(fn, 1);
        } else {
            document.addEventListener("DOMContentLoaded", fn);
        }
    }

    docReady(function () {
        var resultContainer = document.getElementById('qr-reader-results');
        var lastResult, countResults = 0;
        function onScanSuccess(decodedText, decodedResult) {
            if (decodedText !== lastResult) {
                ++countResults;
                lastResult = decodedText;
                // Handle on success condition with the decoded message.
                console.log(`Scan result ${decodedText}`, decodedResult);


    		document.write(

			${decodedText}			

      			`<BODY TEXT="black" BGCOLOR="#D1DADF" LINK="#0000EE" VLINK="#551A8B" ALINK="#FF0000" bgproperties=fixed BACKGROUND=/icons/test.gif><BR><B>
			&nbsp;&nbsp;<IMG SRC=/icons/QR_Code_oliver_lippold_homepage.gif height=50 width=40 ALIGN=middle>&nbsp;&nbsp;<FONT SIZE=+2>QR-Code Scanner</FONT></B><BR>
			<BR><HR><BR><FONT SIZE=-1>


			<TABLE border=0 BACKGROUND=/icons/test2.jpg style=\"box-shadow: 10px 15px 15px gray; border-radius:4px;">
			<TD border=1 width=300pt BACKGROUND=/icons/tabellenkopf5.gif><FONT SIZE=-1>Scanergebnis<BR><B>
			<TR border=1><TD width=200pt FONT SIZE=-1><left>Dein gescannter QR-Code bedeutet:</B><BR>
			<TR><TD width=200pt FONT SIZE=-1><left>&nbsp;</B><BR>
			<TR><TD width=200pt FONT SIZE=-1><left><B>${decodedText}</B><BR>
			<TR><TD width=200pt><CENTER></B><BR><A HREF="/scripts/qrcode_scanner.htm"><FONT SIZE=-1><BR>Zurueck<BR><BR></FONT></A></body>`
    		);



            }
        }

        var html5QrcodeScanner = new Html5QrcodeScanner(
            "qr-reader", { fps: 10, qrbox: 250 });
        html5QrcodeScanner.render(onScanSuccess);
    });
</script>

Any idea ?
Thank's an BR
Oliver

**Comment by Pavel-Sushko** [comment]
What about this:
`decodedText.replace(/(\\n|\\r)/g, "\n")`

**Comment by oliverlippold** [comment]
Hi Pavel,

I have done a test with your code, it does not work :-(. I think the problem is that $(decodedText) has no \r\n etc. , I think there are only " " for space an linefeed. I think the html5-qrcode-scanner has a funktion to print the $(decodedText) in the right format because in the Demo it work, but I can't find a documentation about the return/linefeed:-(. Maybe the creator of the code can implement a better output-funktion for this problem ? ;-) :-)

For more help I say thanks :-)!
BR
Oliver

**Reply by Pavel-Sushko** [reply]
Could you link the example you're referring to?

**Comment by oliverlippold** [comment]
Hi Pavel,

I fond somethink: there is a "Container" named "decodedResult" parallel to decodedText, I can read ${decodedResult.result.format.formatResult} and get the result "QR_CODE", now the question is what name for example !!!!! ${decodedResult.result.text.decodedText} I have to use get the Result-Text from the scan, at the moment I need to know the "feelds from the container "decodedResult".

I hope you know what I meen?
Thanks for help
Oliver
